examples/apps/aml/STATEFULL.md

```diff
@@ -0,0 +1,11 @@
+
+## Changes to make tcf statefull
+1. First we needed to modify *work_order_processor.cpp* and *work_order_processor.h* inside the folder *tc/sgx/trusted_worker_manager/enclave/*. On them we have added a new variable "keepState" that will is parsed from the workOrder json and indicates if we want to keep the WorkloadProcessor that was used on the last execution or we want to create a new one with the state starting from scratch.
+2. Next we have modified the files *workload_processor.cpp* and *workload_processor.h* in folder *common/sgx_workload/*.
+	- We have added 2 new parameters to the function *CreateWorkloadProcessor* (*worker_id* and the newly created *keepState*).
+	- In the cpp file we have added new map called *initialized_processors*. This map will be used to store the state of the already initialized WorkloadProcessors between 2 different workOrders. The key key for that map would be the following one:
```

Comment:
Overall logic looks fine as long as we have single instance of worker handling the workloads.  This should be ideally addressed in spec/design if we intend to make tcf stateful. We need to think through different design options. For example, how will we handle the case when workers and workload processors are in different systems. Currently TCF client uses Rest API model to interact with TCF and hence TCF is stateless.